src/error.rs

```diff
@@ -473,6 +487,10 @@ pub struct WriteConcernError {
     /// A document identifying the write concern setting related to the error.
     #[serde(rename = "errInfo")]
     pub details: Option<Document>,
+
+    /// Labels categorizing the error.
+    #[serde(rename = "errorLabels", default)]
+    pub labels: Vec<String>,
```

Comment:
Error labels are only ever actually returned at the top level of an error response, and it's merely a quirk / bug in the spec tests that they appear in the write concern error. @isabelatkinson actually just put up a PR to fix that: https://github.com/mongodb/specifications/pull/1086. 

For that reason, I don't think we should include public API support for this case, and if need be we can just skip the affected tests until the PR is merged. I think I remember you saying we may have to wait for a serverless change on this too to support the labels being specified in the correct place.

Reply:
The serverless proxy apparently enforces that they appear here; I've removed this field from the public API but I'd prefer to leave it and the related logic in place so the tests can continue to pass.

Reply:
have we confirmed that this behavior is correct/desired on the serverless end? it seems like a weird inconsistency. I think it's fine to have it within the driver right now so that we can run the tests but in the long run it would be nice not to have to maintain a special case like this

Reply:
I'm in the process of following up about this on the serverless side - it looks like this isn't expected behavior.  I'll make sure to send a follow-up PR for the driver (even if it's just a TODO with a bug reference) once that gets sorted out.

Reply:
SGTM to leave this in temporarily until serverless fixes the issue on their end.

Reply:
I've added todos here referencing CLOUDP-105256.